flang/lib/Optimizer/Dialect/FIRType.cpp

```diff
@@ -287,294 +127,16 @@ mlir::Type fir::parseFirType(FIROpsDialect *dialect,
                              mlir::DialectAsmParser &parser) {
   llvm::StringRef typeNameLit;
   if (mlir::failed(parser.parseKeyword(&typeNameLit)))
-    return {};
+    return mlir::Type();
```

Comment:
Prefer `{}` and type inferencing in general. (See use of `auto`.)

Reply:
Fine for me. I was trying to match MLIR example for parser.